test_plans/sycl_ext_oneapi_composite_device.asciidoc

```diff
@@ -0,0 +1,132 @@
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+=== Feature test macro
+
+All of the tests should use `#ifdef SYCL_EXT_ONEAPI_COMPOSITE_DEVICE` so they
+can be skipped if the extension is not supported.
+
+== Tests
+
+Note that some of the sections below closely follow the extension specification
+document and describe tests with narrower scope, which are aimed to check
+specific functionality. There are sections below which describe bigger test
+cases which are intended to check how different elements of the extension work
+together.
+
+=== Enumerating composite devices
+
+==== Through the new `get_composite_devices()` API
+
+This test should check that freestanding function `get_composite_devices()`:
+
+- has the correct signature
+- does not throw an exception
+- returns the same result on subsequent calls
+
+==== Through the new `platform::ext_oneapi_get_composite_devices()` API
+
+This test should take a platform the selected device belongs to and check that
+`platform::ext_oneapi_get_composite_devices()` method:
+
+- has the correct signature
+- does not throw an exception
+- returns the same result on subsequent calls
+
+==== Composite devices are not considered as root devices
+
+The test should ensure that no composite devices are returned through the
+`platform::get_device()` method for a platform the selected device belongs to.
+
+=== New device information descriptors
+
+==== `info::device::component_devices`
+
+The test should check that a query for the information descriptor has correct
+return type.
+
+The test should also check the value returned for the descriptor:
+
+- if the selected device is a component device, an empty vector is expected
+- otherwise, returned vector is expected to be non-empty and each its element
+  should be additionally checked:
+  - each component device should have the `ext_oneapi_is_component` aspect
+  - each component device should return the selected device through
+    `info::device::composite_device` query
+
+==== `info::device::composite_device`
+
+The test should check that a query for the information descriptor has correct
+return type.
+
+The test should also check the value returned for the descriptor:
+
+- if the selected device is a component device, then a synchronous `exception`
+  with `errc::invalid` error code is expected
+- otherwise, returned device should have the `ext_oneapi_is_composite` aspect
+
+=== New device aspects
+
+==== `ext_oneapi_is_composite`
+
+The test should check that if the selected device has the aspect, then the
+device should be present in a list returned by both `get_composite_devices()`
+and `platform::ext_oneapi_get_composite_devices()`.
+
+==== `ext_oneapi_is_component`
+
+The test should check that if the selected device has the aspect, then the
+device should *not* be present in a list returned by either
+`get_composite_devices()` or `platform::ext_oneapi_get_composite_devices()`.
```

Comment:
It would be more correct to do this test:

> If the selected device does not have the `ext_oneapi_is_composite` aspect, then the device should *not* be present in the list returned by either `get_composite_devices()` or `platform::ext_oneapi_get_composite_devices()`.

(Remember, there could be devices that are neither composite nor component devices.)

However, you already perform this test above under the section "Composite devices are not considered as root devices".  Therefore, I think the test described here is really just a duplicate.

Reply:
9e548b8938ca376b49e847e3d23de872befbc280 rewrites tests for aspect in accordance of how they are implemented (#876):
- there are basic tests to check aspects consistency which use `get_composite_devices()` and `platform::ext_oneapi_get_composite_devices()` results
- there is a test specifically for component devices to check how their partitioning works

Rest of the things related to the aspects and their interaction with other extension functionality is covered by other tests, I think